util/network-devp2p/src/discovery.rs

```diff
@@ -200,41 +230,53 @@ impl<'a> Discovery<'a> {
 		}
 	}
 
-	fn update_node(&mut self, e: NodeEntry) -> Option<TableUpdates> {
-		trace!(target: "discovery", "Inserting {:?}", &e);
+	fn update_bucket_record(&mut self, e: NodeEntry) -> Result<(), BucketError> {
```

Comment:
Personal preference on returning an 'Option' here and not having 'BucketError' field (If I make no mistake BucketError::Ourself got same behavior as Ok except for a debug trace). But it is really only a personal preference (does not really matter).

Reply:
Actually it's not true, and [discovery.rs:259](https://github.com/paritytech/parity-ethereum/pull/10305/files#diff-3193558172eb0ec141a7afd53b7a1118R259) we differentiate depending on the nature of the `BucketError`.
We can, theoretically, use `Option` to indicate the specific kind of error we want to handle there — but I don't think that will be easier to read than what I have now. (It's not obvious why would some errors while updating end up with `Some` and another with `None`.)

Reply:
I meant making Ok and BucketError::ourselve the same variant (it requires dropping a debug trace), but as I said it is just a matter a personal preference.